### PR TITLE
Update code to use ```Meeting``` Attribute

### LINKED ISSUE #52: Implement add functionality for Meeting

### PR DESCRIPTION
**Pull request by Pluiexo**
Meeting has been implemented in a way that is not part of the Persons constructor, this is to reduce the amount of coupling there is between ```Person```  and ```Meeting```. Which would allow for better expansions in the future etc, like having the same meeting with multiple  ```Person``` etc.

For simplification, the default  list of contacts will not contain any meetings assigned to them

The UI list for showing meetings is very rudimentary for now and it will be fixed in #53 

Closes #52

### CODE REVIEW

File: src/main/java/staffconnect/logic/commands/AddMeetingCommand.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+     */
+    private static Person addMeetingToPerson(Person personToEdit, Meeting meeting) {
+        assert personToEdit != null;
+
+        Name currentName = personToEdit.getName();
+        Phone currentPhone = personToEdit.getPhone();
+        Email currentEmail = personToEdit.getEmail();
+        Venue currentVenue = personToEdit.getVenue();
+        Module currentModule = personToEdit.getModule();
+        Faculty currentFaculty = personToEdit.getFaculty();
```
Comment (tsulim) [inline src/main/java/staffconnect/logic/commands/AddMeetingCommand.java]: Minor nit to fix. `Module` -> `Faculty` -> `Venue`

File: src/main/java/staffconnect/logic/commands/AddMeetingCommand.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        Set<Tag> currentTags = personToEdit.getTags();
+        Set<Meeting> currentMeetings = new HashSet<>(personToEdit.getMeetings()); //to reduce coupling with Person
+        Set<Availability> currentAvailability = personToEdit.getAvailabilities();
```
Comment (tsulim) [inline src/main/java/staffconnect/logic/commands/AddMeetingCommand.java]: I think it might be more logical to have it as `Availability` -> `Meeting`. What's your take on this? 🤔

File: src/main/java/staffconnect/model/person/Person.java [hunk 3]
```diff
@@ -29,6 +30,8 @@ public class Person {
     private final Set<Tag> tags = new HashSet<>();
     private final Set<Availability> availabilities = new HashSet<>();
 
+    private Set<Meeting> meetings = new HashSet<>();
```
Comment (tsulim) [inline src/main/java/staffconnect/model/person/Person.java]: Should this be `final`?

File: src/main/java/staffconnect/model/person/Person.java [hunk 4]
```diff
@@ -45,10 +48,6 @@ public Person(Name name, Phone phone, Email email, Module module, Faculty facult
         this.availabilities.addAll(availabilities);
     }
 
-    public Name getName() {
-        return name;
-    }
-
```
Comment (tsulim) [inline src/main/java/staffconnect/model/person/Person.java]: Was the shift for `getName()` intentional? If so, what was your reasoning behind doing so?

File: src/main/java/staffconnect/model/person/Person.java [hunk 5]
```diff
@@ -95,7 +105,18 @@ public boolean isSamePerson(Person otherPerson) {
         }
 
         return otherPerson != null
-                && otherPerson.getName().equals(getName());
+            && otherPerson.getName().equals(getName());
```
Comment (tsulim) [inline src/main/java/staffconnect/model/person/Person.java]: Not sure if this is common or not, but I personally think that this should be reverted to the original indent. Any opinions or thoughts on this?

File: src/main/java/staffconnect/storage/JsonAdaptedPerson.java [hunk 6]
```diff
@@ -44,7 +47,9 @@ public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone
             @JsonProperty("email") String email, @JsonProperty("module") String module,
             @JsonProperty("faculty") String faculty, @JsonProperty("venue") String venue,
             @JsonProperty("tags") List<JsonAdaptedTag> tags,
-            @JsonProperty("availabilities") List<JsonAdaptedAvailability> availabilities) {
+            @JsonProperty("availabilities") List<JsonAdaptedAvailability> availabilities,
+                             @JsonProperty("meetings") List<JsonAdaptedMeeting> meetings) {
```
Comment (tsulim) [inline src/main/java/staffconnect/storage/JsonAdaptedPerson.java]: Perhaps the indentation for `@JsonProperty("meetings") ...` should be the same level as the ones above it?

File: src/main/java/staffconnect/storage/JsonAdaptedPerson.java [hunk 7]
```diff
@@ -70,11 +78,16 @@ public JsonAdaptedPerson(Person source) {
...
+                        .map(JsonAdaptedTag::new)
+                        .collect(Collectors.toList()));
+
         availabilities.addAll(source.getAvailabilities().stream()
             .map(JsonAdaptedAvailability::new)
             .collect(Collectors.toList()));
+
+        meetings.addAll(source.getMeetings().stream()
+                            .map(JsonAdaptedMeeting::new)
+                            .collect(Collectors.toList()));
```
Comment (tsulim) [inline src/main/java/staffconnect/storage/JsonAdaptedPerson.java]: Minor nit: Standardize indentation level

File: src/main/java/staffconnect/ui/MeetingsCard.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    private Label date;
+
+    /**
+     * Creates a {@code MeetingCode} with the given {@code Meeting} and index to display.
+     */
+    public MeetingsCard(Meeting meeting) {
+        super(FXML);
+        description.setText(meeting.getDescription().toString());
+        date.setText(meeting.getStartDate().toString());
+    }
```
Comment (tsulim) [inline src/main/java/staffconnect/ui/MeetingsCard.java]: For future implementation, I think it might be better to add a display index so as to be able to reference it. E.g. edit meeting, delete meeting.

File: src/main/java/staffconnect/ui/PersonCard.java [hunk 9]
```diff
@@ -61,10 +67,14 @@ public PersonCard(Person person, int displayedIndex) {
...
-                .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
+            .sorted(Comparator.comparing(tag -> tag.tagName))
+            .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
+
         person.getAvailabilities().stream()
                 .sorted(Comparator.comparing(availability -> availability.value))
                 .forEach(availability -> availabilities.getChildren().add(new Label(availability.value)));
+
+        person.getMeetings().stream().sorted(Comparator.comparing(meeting -> meeting.getStartDate().getDateTime()))
+            .forEach(meeting -> meetingsContainer.getChildren().add(new MeetingsCard(meeting).getRoot()));
```
Comment (tsulim) [inline src/main/java/staffconnect/ui/PersonCard.java]: Minor nit: Standardize indentation level

File: src/main/java/staffconnect/ui/MeetingsCard.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+public class MeetingsCard extends UiPart<Region> {
+
+    private static final String FXML = "MeetingsListCard.fxml";
+    @FXML
+    private Label description;
+    @FXML
+    private Label date;
+
+    /**
+     * Creates a {@code MeetingCode} with the given {@code Meeting} and index to display.
```
Comment (Pluiexo) [inline src/main/java/staffconnect/ui/MeetingsCard.java:23]: Don't mind this typo will fix in the UI revamp

File: src/test/java/staffconnect/logic/commands/AddMeetingCommandTest.java [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    private Person buildPerson() {
+        Person pickPerson = TEST_MODEL.getFilteredPersonList().get(0);
+        Person validPerson = new Person(pickPerson.getName(), pickPerson.getPhone(), pickPerson.getEmail(),
+                                        pickPerson.getModule(), pickPerson.getFaculty(), pickPerson.getVenue(),
+                                        pickPerson.getTags(), pickPerson.getAvailabilities());
+        validPerson.setMeetings(new HashSet<>(List.of(VALID_MEETING)));
+        return validPerson;
+    }
```
Comment (tsulim) [inline src/test/java/staffconnect/logic/commands/AddMeetingCommandTest.java]: Should this method be above all tests?

File: src/test/java/staffconnect/logic/commands/AddMeetingCommandTest.java [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+
+    private static final Model TEST_MODEL = new ModelManager(getTypicalStaffBook(), new UserPrefs());
+
+
+    @Test
+    public void execute_allFieldsValid_success() {
+        Person validPerson = buildPerson();
+        AddMeetingCommand addMeetingCommand = new AddMeetingCommand(INDEX_FIRST_PERSON, VALID_MEETING);
+        String expectedMessage = String.format(AddMeetingCommand.MESSAGE_SUCCESS, Messages.format(VALID_MEETING));
+        Model expectedModel = new ModelManager(new StaffBook(TEST_MODEL.getStaffBook()), new UserPrefs());
```
Comment (tsulim) [inline src/test/java/staffconnect/logic/commands/AddMeetingCommandTest.java]: Perhaps you can enhance readability by adding a blank line between this 3 steps?

File: src/test/java/staffconnect/logic/commands/AddMeetingCommandTest.java [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+                                        pickPerson.getModule(), pickPerson.getFaculty(), pickPerson.getVenue(),
+                                        pickPerson.getTags(), pickPerson.getAvailabilities());
+        validPerson.setMeetings(new HashSet<>(List.of(VALID_MEETING)));
+        return validPerson;
+    }
+
+    @Test
+    public void execute_duplicateMeeting_failure() {
+        Person firstPerson = TEST_MODEL.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        firstPerson.setMeetings(new HashSet<>(List.of(VALID_MEETING)));
```
Comment (tsulim) [inline src/test/java/staffconnect/logic/commands/AddMeetingCommandTest.java]: Any reason why you've written this instead of using `buildPerson()`?

File: src/test/java/staffconnect/logic/commands/AddMeetingCommandTest.java [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+
+        AddMeetingCommand addMeetingCommand = new AddMeetingCommand(INDEX_FIRST_PERSON, VALID_MEETING);
+
+        Model duplicateModel = new ModelManager(new StaffBook(TEST_MODEL.getStaffBook()), new UserPrefs());
+        duplicateModel.setPerson(TEST_MODEL.getFilteredPersonList().get(0), firstPerson);
+
+        assertCommandFailure(addMeetingCommand, duplicateModel, AddMeetingCommand.MESSAGE_DUPLICATE_MEETING);
```
Comment (tsulim) [inline src/test/java/staffconnect/logic/commands/AddMeetingCommandTest.java:73]: Any reason why you did it this way instead of using `TEST_MODEL`?

File: src/test/java/staffconnect/logic/commands/CommandTestUtil.java [hunk 15]
```diff
@@ -48,6 +53,13 @@ public class CommandTestUtil {
 
 
 
+    public static final String VALID_MEETING_DESCRIPTION = "Meet for midterms";
+
+    public static final String VALID_DIFF_MEETING_DESCRIPTION = "Meet for finals";
+    public static final String VALID_MEETING_DATE = "12/04/2023 18:00";
+    public static final String VALID_DIFF_MEETING_DATE = "15/04/2024 15:00";
```
Comment (tsulim) [inline src/test/java/staffconnect/logic/commands/CommandTestUtil.java]: Perhaps the naming convention should be standardized?
For example, this portion of the code is `VALID_ATTRIBUTE_<something that differentiates>` so it could be `VALID_DESCRIPTION_AMY`. Don't have to follow the naming of `AMY` and `BOB` but it is one way of differentiating the variables
This is applicable to other places too.

File: src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+
+    private static final String MESSAGE_INVALID_FORMAT =
+        String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddMeetingCommand.MESSAGE_USAGE);
+
+    private final AddMeetingCommandParser parser = new AddMeetingCommandParser();
+
+    @Test
+    public void parse_missingParts_failure() {
+        // no index specified
+        assertParseFailure(parser, MEETING_MISSING_INDEX, MESSAGE_INVALID_FORMAT);
```
Comment (tsulim) [inline src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java]: Any reason why you use `MEETING_MISSING_INDEX` instead of having `VALID_MEETING_DESC` in `CommandTestUtil` which could be reused in other test files?

File: src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+
+    @Test
+    public void parse_invalidValue_failure() {
+
+        // invalid description followed by valid date
+        assertParseFailure(parser, "1" + INVALID_DESCRIPTION + MEETDATE_STARTDATE,
+                           Description.MESSAGE_CONSTRAINTS);
+        // valid description followed by invalid date
+        assertParseFailure(parser, "1" + DESC_MIDTERM + INVALID_STARTDATE, MeetDateTime.MESSAGE_CONSTRAINTS);
+    }
```
Comment (tsulim) [inline src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java]: Perhaps you could add in a test case for `two invalid values, only first invalid value reported` and `non-empty preamble` here

File: src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+
+        AddMeetingCommand expectedCommand = new AddMeetingCommand(targetIndex, VALID_MEETING);
+
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    @Test
+    public void parse_multipleRepeatedFields_failure() {
+        // More extensive testing of duplicate parameter detections is done in
+        // AddCommandParserTest#parse_repeatedNonTagValue_failure()
```
Comment (tsulim) [inline src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java]: Should this be removed?

File: src/test/java/staffconnect/storage/JsonAdaptedPersonTest.java [hunk 19]
```diff
@@ -29,19 +30,31 @@ public class JsonAdaptedPersonTest {
...
+
+    private static final String VALID_MEETING_DATE = "12-04-2023 18:00";
+
+    private static final JsonAdaptedMeeting VALID_MEETING_JSON =
+        new JsonAdaptedMeeting(VALID_MEETING_DESCRIPTION, VALID_MEETING_DATE);
+    private static final List<JsonAdaptedMeeting> VALID_MEETING =
+        Stream.of(VALID_MEETING_JSON).collect(Collectors.toList());
+
+    private static final List<JsonAdaptedTag> VALID_TAGS =
+        BENSON.getTags().stream().map(JsonAdaptedTag::new).collect(Collectors.toList());
```
Comment (tsulim) [inline src/test/java/staffconnect/storage/JsonAdaptedPersonTest.java:43]: Any reason for the shift in code for `VALID_TAGS`?

File: src/test/java/staffconnect/storage/JsonAdaptedPersonTest.java [hunk 20]
```diff
@@ -150,16 +175,37 @@ public void toModelType_invalidTags_throwsIllegalValueException() {
...
 
+    @Test
+    public void toModelType_invalidMeetingDate_throwsIllegalValueException() {
+        List<JsonAdaptedMeeting> invalidMeeting = new ArrayList<>(VALID_MEETING);
+        invalidMeeting.add(new JsonAdaptedMeeting(VALID_MEETING_DESCRIPTION, INVALID_MEETING_DATE));
+        JsonAdaptedPerson person =
+            new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, VALID_EMAIL, VALID_MODULE,
+                                  VALID_FACULTY, VALID_VENUE, VALID_TAGS, VALID_AVAILABILITIES, invalidMeeting);
+        assertThrows(IllegalValueException.class, person::toModelType);
+    }
```
Comment (tsulim) [inline src/test/java/staffconnect/storage/JsonAdaptedPersonTest.java]: May I request that you shift this portion below `Availability`? Thank you

File: src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java
Comment (tsulim) [inline src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java:1]: I think you can add a test on missing fields in `AddMeetingCommandParserTest`

**Review by tsulim — COMMENTED** [review]
Overall, there's quite a number of things to address, some of which are nits. I think you might need to be more careful when fixing merge conflicts as I noticed that some of the codes are shifted out of their original place/order.
Thank you for making the deadline. If there are any questions/doubts, feel free to ask. 👍

File: src/main/java/staffconnect/model/person/Person.java (hunk 3 above)
Comment (Pluiexo) [reply src/main/java/staffconnect/model/person/Person.java]: This can't be ```final``` because of the way I assign a new set of ```meeting``` to a person. This is to allow for future expansions and ```meeting``` to be mutable and change regularly without creating redundant immutable copies

Comment (Pluiexo) [reply src/main/java/staffconnect/model/person/Person.java]: But I agree that the rest of the fields like availabilities, name etc.. should remain as final because, they are not updated as frequently as ```meeting```

File: src/main/java/staffconnect/model/person/Person.java (hunk 4 above)
Comment (Pluiexo) [reply src/main/java/staffconnect/model/person/Person.java]: Alot of the code order got messed up by the incoming merge, and I do apologies if I missed out a few

File: src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java
Comment (Pluiexo) [reply src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java:1]: Missing Fields for ```AddMeetingCommandParserTest``` is in there as ```parse_missingParts_failure```. Any missing fields in the prefix is already tested by the corresponding ```Description``` or ```MeetDateTime```. Also any missing attributes in the prefix will cause ```AddMeetingCommandParserTest``` to never reach any assert statements because the error is thrown in ```Description``` and ```MeetDateTime```;

Comment (Pluiexo) [reply src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java:1]: But I will add more testcases to ```parse_missingParts_failure```

File: src/test/java/staffconnect/storage/JsonAdaptedPersonTest.java (hunk 19 above)
Comment (Pluiexo) [reply src/test/java/staffconnect/storage/JsonAdaptedPersonTest.java:43]: Shifted by merge conflict

File: src/main/java/staffconnect/storage/JsonAdaptedPerson.java (hunk 6 above)
Comment (Pluiexo) [reply src/main/java/staffconnect/storage/JsonAdaptedPerson.java]: Fixed

File: src/main/java/staffconnect/model/person/Person.java (hunk 5 above)
Comment (Pluiexo) [reply src/main/java/staffconnect/model/person/Person.java]: This was messed up by the merge too, fixed

File: src/main/java/staffconnect/storage/JsonAdaptedPerson.java (hunk 7 above)
Comment (Pluiexo) [reply src/main/java/staffconnect/storage/JsonAdaptedPerson.java]: Fixed, and retuned my auto formatter settings

File: src/main/java/staffconnect/logic/commands/AddMeetingCommand.java (hunk 1 above)
Comment (Pluiexo) [reply src/main/java/staffconnect/logic/commands/AddMeetingCommand.java]: Fixed

File: src/main/java/staffconnect/logic/commands/AddMeetingCommand.java (hunk 2 above)
Comment (Pluiexo) [reply src/main/java/staffconnect/logic/commands/AddMeetingCommand.java]: Fixed

File: src/main/java/staffconnect/model/person/Person.java (hunk 3 above)
Comment (tsulim) [reply src/main/java/staffconnect/model/person/Person.java:33]: I understand. Thanks for clarifying! 👍

File: src/test/java/staffconnect/logic/commands/AddMeetingCommandTest.java (hunk 14 above)
Comment (Pluiexo) [reply src/test/java/staffconnect/logic/commands/AddMeetingCommandTest.java:73]: That is to create a ```duplicateModel``` which is a copy of ```TEST_MODEL``` with the first meeting already added to the first person.Then the same meeting will be added with the command by the assert command . ```TEST_MODEL``` does not contain any meetings assigned

File: src/main/java/staffconnect/ui/PersonCard.java (hunk 9 above)
Comment (Pluiexo) [reply src/main/java/staffconnect/ui/PersonCard.java]: Fixed I had a error in my auto formatter

File: src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java (hunk 16 above)
Comment (Pluiexo) [reply src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java]: Fixed

File: src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java (hunk 17 above)
Comment (Pluiexo) [reply src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java:66]: Added ```two invalid values``` and ```non-empty preamble```, I don't think ```first invalid value reported ``` is applicable in my case. Because it will fall into its children class error, or and invalid command if either one of the values are invalid

File: src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java (hunk 18 above)
Comment (Pluiexo) [reply src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java]: Fixed

File: src/test/java/staffconnect/storage/JsonAdaptedPersonTest.java (hunk 20 above)
Comment (Pluiexo) [reply src/test/java/staffconnect/storage/JsonAdaptedPersonTest.java]: Fixed

File: src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java (hunk 17 above)
Comment (Pluiexo) [reply src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java]: Fixed

File: src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java
Comment (tsulim) [reply src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java:1]: > Missing Fields for `AddMeetingCommandParserTest` is in there as `parse_missingParts_failure`. Any missing fields in the prefix is already tested by the corresponding `Description` or `MeetDateTime`. Also any missing attributes in the prefix will cause `AddMeetingCommandParserTest` to never reach any assert statements because the error is thrown in `Description` and `MeetDateTime`;

From what I've reviewed, I believe I did not see any tests regarding missing fields in prefix.

File: src/main/java/staffconnect/model/person/Person.java (hunk 3 above)
Comment (whitesnowx) [reply src/main/java/staffconnect/model/person/Person.java]: Correct me if im wrong, your implementation of `meeting` is intentionally different from availability/tags because you want to update the meeting attribute without creating a new person each time? 

imo, I think it's not worth differing in implementation because `final` is usually used to protect the code (security and safety) over the time taken to create a new person.

This is debatable but for time concern, can be moved to a different iteration/pr for changes

Comment (tsulim) [reply src/main/java/staffconnect/model/person/Person.java]: I understand that you require meeting to be mutable. However, I think there might have been a misunderstanding of `final`. `final` simply makes the object _reference_ unchangeable. The object it points to is not immutable by doing so, thus if `meetings` is set as `final`, it can never refer to another object, but the object it refers to may change state.

Hopefully this addresses your concern.

File: src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java
Comment (Pluiexo) [reply src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java:1]: May I clarify what does your missing fields refer to? Are they missing prefix, or missing details in the prefix but the prefix is there?

File: src/main/java/staffconnect/ui/MeetingsCard.java (hunk 8 above)
Comment (Pluiexo) [reply src/main/java/staffconnect/ui/MeetingsCard.java]: Give me a moment for this, the current implementation for it makes it difficult to get the index for the meeting.
As originally I was conflicted to use a listView here, as it adds substantial amount of complexity, but right now I am testing it by changing it to a listview

File: src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java
Comment (tsulim) [reply src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java:1]: It refers to missing prefix when the user input valid `description` and `meetdatetime`. (i.e., `meeting 1 meet for midterms s/12-04-2023 1800` and `meeting 1 d/meet for midterms 12-04-2023 1800`).
Please let me know if it was already coded and that I may have missed it out in the code.

File: src/main/java/staffconnect/model/meeting/MeetDateTime.java
Comment (whitesnowx) [inline src/main/java/staffconnect/model/meeting/MeetDateTime.java:1]: `public static final String VALIDATION_REGEX = "\\d{2}/\\d{2}/\\d{4} \\d{2}:\\d{2}";`

There is a bug when entering values that fall out of valid date and time ranges. E.g:
- meeting 1 d/test1 s/99/11/1111 11:11
- meeting 1 d/test2 s/11/11/1111 99:11

### Can try:
Amend `isValidMeetDateTime(String test)` to catch (DateTimeParseException e) and handle that exception accordingly

_missed this during implementation of MeetDateTime class_

**Review by whitesnowx — COMMENTED** [review]
- Implementation might not be optimal. Probably worth discussing for a separate iteration.
- Formatting issues mentioned in @tsulim review noted 'Fixed' are not updated? 
- **Bug: on invalid date/time (regex/exception in MeetDateTime.java)** <-- waiting on this
- Reminder to update UserGuide in the next iteration
- Nice implementation of the UI adaptation of meetings

File: src/main/java/staffconnect/model/meeting/MeetDateTime.java
Comment (Pluiexo) [reply src/main/java/staffconnect/model/meeting/MeetDateTime.java:1]: > `public static final String VALIDATION_REGEX = "\\d{2}/\\d{2}/\\d{4} \\d{2}:\\d{2}";`
> 
> There is a bug when entering values that fall out of valid date and time ranges. E.g:
> 
> * meeting 1 d/test1 s/99/11/1111 11:11
> * meeting 1 d/test2 s/11/11/1111 99:11
> 
> ### Can try:
> Amend `isValidMeetDateTime(String test)` to catch (DateTimeParseException e) and handle that exception accordingly
> 
> _missed this during implementation of MeetDateTime class_

Thanks for spotting this out, will fix that bug. Though this should be related to the previous pull request and not this one!

Comment (Pluiexo) [reply src/main/java/staffconnect/model/meeting/MeetDateTime.java:1]: Yup you are right for the first part , ```Meeting ``` is intentionally made mutable and weakly coupled with ``Person```. This is to allow easy extraction of the implementation if we choose to separate it and implement more advance features on it! Also we do not want to identify ```Person``` by ```Meeting``` right now. That means to say that if two ```Person``` have the same attributes for everything except having different meetings, they should identify as the same ```Person``` and not different

Comment (Pluiexo) [reply src/main/java/staffconnect/model/meeting/MeetDateTime.java:1]: The first part has been changed to final

**Comment by Pluiexo** [comment]
Fix changes had been made:

- [x] Convert Vbox UI to listView to show index of meetings
- [x] Convert meetings in ```Person``` to final
- [x] Fix formatting issues
- [x] Added more testcases to AddMeetingCommandParser ```Invalid Prefix```
- [x] Fix MeetDateTime invalid dateTime values like 99/01/2023 00:99

Also reply to this comment if there are anything missing since the rest are outdated

**Review by tsulim — APPROVED** [review]
LGTM! 👍
